modules/ref/pages/client-settings.adoc

```diff
@@ -96,524 +123,523 @@ By default the client will use a simple matching heuristic to determine which se
 +
-Default: `10s`
+Parameter: `config_idle_redial_timeout: Optional[timedelta]`
 +
-System Property: `com.couchbase.env.timeout.disconnectTimeout`
+Default:  `None`
 +
-The disconnect timeout is used when a Cluster is disconnected and if not overridden by a custom timeout.
-A timeout is applied here always to make sure that your code does not get stuck at shutdown.
-The default should provide enough time to drain all outstanding operations properly, but make sure to adapt this timeout to fit your application requirements.
+// TODO: better description
+Config Idle Redial Timeout.
 
-Name: *Management Timeout*::
-Parameter: `TimeoutConfig.managementTimeout(Duration)`
+=== Compression Options
+
+Name: *Enabling Compression*::
+Options Class: `ClusterOptions`
 +
-Default: `75s`
+Parameter: `enable_compression: Optional[bool]`
 +
-System Property: `com.couchbase.env.timeout.managementTimeout`
+Default:  `True`
 +
-The management timeout is used on all cluster management APIs (BucketManager, UserManager, CollectionManager, QueryIndexManager, etc.) if not overridden by a custom timeout.
-The default is quite high because some operations (such as flushing a bucket, for example) might take a long time.
+If enabled, the client will compress documents before they are sent to Couchbase Server.
+// If this is set to `False`, the other CompressionExample settings have no effect.
+// TODO:  compression_min_size, compression_min_ratio
 
-== Diagnostics Options
+=== Tracing Options
 
-The client can periodically log diagnostic information about server connectivity.
+// TODO:  all options need better descriptions + default values
+Tracing settings are configured using an instance of `ClusterTracingOptions` or passed in as keyword arguments.
 
-Diagnostics settings are represented by the Java class `DiagnosticsConfig`.
-The associated `ClusterEnvironement.Builder` method is called `diagnosticsConfig`.
+.Creating a new `ClusterTracingOptions` instance:
 
-.Template for configuring diagnostics settings
-[source,java]
+[source,python]
 ----
-ClusterEnvironment env = ClusterEnvironment.builder()
-    .diagnosticsConfig(DiagnosticsConfig
-        .enabled(true)
-        ...)
-    .build()
+tracing_options=ClusterTimeoutOptions(tracing_threshold_kv=timedelta(milliseconds=5), tracing_threshold_query=timedelta(milliseconds=10))
+options=ClusterOptions(PasswordAuthenticator('username', 'password'), tracing_options=tracing_options)
+cluster.connect('couchbase://host',options)
 ----
 
-In addition to the connection string, the [.api]`couchbase_core.client.Client` object offers a few object attributes which can be used to configure settings during runtime:
-
-* `timeout` for key-value operation timeouts (equivalent to `operation_timeout` in the connection string)
-* `views_timeout` to adjust the timeout for MapReduce queries
-* `n1ql_timeout` to adjust the timeout for N1QL querties
+Name: *Tracing Enabled*::
+Options Class: `ClusterOptions`
```

Comment:
Just to check, here and line 363 are `ClusterOptions`, whereas following are `ClusterTracingOptions`, is this intentional?

Reply:
At the moment `enable_tracing` and `enable_metrics` are actually in `ClusterOptions` and not `ClusterTracingOptions`.  I suppose b/c those are more 'global' options.  Maybe to avoid confusion we can move those to general options?  I think to align w/ 3.x we need to keep them in `ClusterOptions`.  Although, since we are dealing w/ Python dict object, we might be able to shift some things around though.

Reply:
No need to trigger a change in SDK behaviour as long as that's documented as intended! 👍